crates/asm/subprotocols/checkpoint/src/handler.rs

```diff
@@ -26,13 +28,14 @@ use crate::{
 /// Steps:
 /// 1. Extract signed checkpoint from envelope
 /// 2. Verify signature
-/// 3. Validate state transitions (epoch, L1/L2 progression)
-/// 4. Validate start values match expected state
-/// 5. Get manifest hashes from auxiliary data
-/// 6. Construct claim and verify proof
-/// 7. Update state with verified checkpoint
-/// 8. Forward withdrawal intents to bridge
-/// 9. Emit checkpoint update log
+/// 3. Validate start values match expected state
+/// 4. Validate state transitions (epoch, L1/L2 progression)
+/// 5. Validate pre-state root matches expected state
```

Comment:
You added the pre-state root verification, but it's in the same step.
```suggestion
/// 4. Validate state transitions (epoch, L1/L2 progression, pre-state root)
```
change the following to start from 5.